src/box/session.cc

```diff
@@ -375,3 +377,71 @@ generic_session_sync(struct session *session)
 	(void) session;
 	return 0;
 }
+
+API_EXPORT int
+box_session_id()
+{
+	return current_session()->id;
+}
+
+API_EXPORT bool
+box_session_exists(uint64_t sid)
+{
+	return (session_find(sid) != NULL);
+}
+
+API_EXPORT const char *
+box_session_peer(uint64_t sid)
+{
+	struct session *session = session_find(sid);
+	if (session == NULL)
+		session = current_session();
+	if (session == NULL)
```

Comment:
How is it possible second time? `current_session()` is not exception-safe, so if it couldn't create a session, it throws. Not returns `NULL`. Which is actually an issue existing before this patch, and because of which even `box_session_id()` can fail. Probably we must add `panic()` in case a session couldn't be created. In a separate commit of course.